x-pack/solutions/security/plugins/security_solution/public/common/components/visualization_actions/use_lens_attributes.tsx

```diff
@@ -123,7 +119,28 @@ export const useLensAttributes = ({
     }
 
     const indexFilters = hasAdHocDataViews ? [] : getIndexFilters(selectedPatterns);
-    const query = esqlQuery ? { esql: esqlQuery } : globalQuery;
+    const query = esql ? { esql } : globalQuery;
+
+    const queryFilters =
+      applyGlobalQueriesAndFilters && esql
+        ? [
+            {
+              meta: {
+                alias: null,
+                disabled: false,
+                key: 'globalFilterKey',
+                negate: false,
+                params: {},
+                type: 'string',
+              },
+              query: {
+                bool: {
+                  filter: globalFilterQuery,
+                },
+              },
+            },
+          ]
+        : [...(applyGlobalQueriesAndFilters ? filters : [])];
```

Comment:
NIT: Nested ternaries are harder to read, wdyt about:

```suggestion
const queryFilters = (() => {
  if (!applyGlobalQueriesAndFilters) return [];
  
  if (esql) {
    return [{
      meta: {
        alias: null,
        disabled: false,
        key: 'globalFilterKey',
        negate: false,
        params: {},
        type: 'string',
      },
      query: {
        bool: {
          filter: globalFilterQuery,
        },
      },
    }];
  }
  
  return filters
})();
```

Reply:
Thank you for the comment!